x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -752,9 +753,22 @@ private static void addMetricFieldMapping(final XContentBuilder builder, final S
                 builder.field(fieldProperty, fieldProperties.get(fieldProperty));
             }
         } else {
-            final String[] supportedAggsArray = metricType.supportedAggs();
+            List<String> supportedAggs = List.of(metricType.supportedAggs());
             // We choose max as the default metric
-            final String defaultMetric = List.of(supportedAggsArray).contains("max") ? "max" : supportedAggsArray[0];
+            String defaultMetric = supportedAggs.contains("max") ? "max" : supportedAggs.get(0);
```

Comment:
Maybe isolate the logic that extracts default metrics in a (static?) method? Then we can maybe add a unit test for this in `TransportDownsampleActionTests`.